rfds/116-Handling-Old-Comments/index.adoc

```diff
@@ -0,0 +1,391 @@
+
+Delete :: Remove previous comments entirely once new ones are posted.
+* Pros: Cleanest PR view.
+* Cons: Loss of historical context, users can't manually fix stuff if something bad happens.
+
+TODO: Discouse about what to do about deleted `apply` comments
+
+Update (Re-use Comments) :: Link comments to a specific directory (dirspace) via metadata in the work_manifest, and update them in-place.
+* Pros: Clean UI, less noisy. Once inplemented, summaries are just a corollary of updates.
+* Cons: Requires properly tracking comment IDs to `work_manifests`. Hardest one to implement.
+
+NOTE: The update strategy was initially considered, but due the its complexity, its only being partially described in this RFD.
+
+Besides the following 4 strategies, there are some extra properties to consider:
+
+* Once a comment is posted as a `plan`/`apply`, no matter the comment strategy, it *should not* be replaced/deleted by the output of an opposite command (i.e. a "plan" output cannot be updated in-place to an "apply").
+
+image::00_comment_plan.png[Plan Comment]
+
+* We always keep the latest `plan`/`apply` of a particular _dispace_.
+* An addendum of the previous property, the output of a `plan`/`apply` from a dirspace `A` should not change the output of a `plan`/`apply` from a dirspace `B`.
+* Also, we keep the latest `plan`/`apply` until all of its contents have been re-generated. Other comments are handled accordinly by the selected strategy.
+* We always reserve a little bit of space to update older comments (to provide admonitions).
+* Keep the error message around until it's no longer needed (TODO: properly describe conditions later).
+
+==== Terrateam Configuration
+
+The following is a proposal of how that would look like as a top level key:
+
+[source,yaml]
+----
+# Default configuration that solves 80% of customer pain
+# TODO: `notifications` is a temporary name, still looking for somethig better.
+notifications:
+  policies:
+    # Default configuration
+    - tag_query: ""
+      # append, delete, minimize, update
+      update_strategy: minimize
+
+    # Extra suggested options (for a future RFD)
+    - tag_query: ""
+      group_id: "A"
+      # append, delete, minimize, update
+      update_strategy: minimize
+      # enabled, disabled
+      commit_status: "disabled"
+----
+
+===== Primitives
+
+Every `update_strategy` will obey a certain set of primitive operations under certain limitations, _L_ represents a limit of how much content we can put in a single comment (different platforms have different limits).
+
+TIP: For GitHub, a _L_ is currently set as 65 kb, for GitLab it's 1Mb.
+TODO: Link to the docs with the oficial values.
+
+These operations will work independent of the chosen strategy:
+
+Break Large Single Outputs :: How to proceed if the output of a run is larger than a single comment.
```

Comment:
How does this interact with multiple dirspace outputs that are split over multiple comments?

Reply:
This is handled in the next operation, if you have outputs where `(output_id, index)` looks like `(1000, 1), (1000, 2), (1000, 3)`, and another dirspace that generates `(1001, 1), (1001, 2)` will be sorted separately and close together.

Now, does this leave room for a another "grouping" operation? Definitely.